lib/mixlib/install/generator/powershell.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Mixlib
+  class Install
+    class Generator
+      class PowerShell
+        attr_reader :options
+
+        def initialize(options)
+          @options = options
+        end
+
+        def install_command
+          install_command = []
+          install_command << "new-module -name Omnitruck -scriptblock {"
+          install_command << get_script(:helpers)
+          if options.for_artifactory?
+            raise "not implemented yet"
+            # install_command << get_script(:get_project_metadata_for_artifactory)
+          else
+            install_command << get_script(:get_project_metadata)
+          end
+          install_command << get_script(:install_project)
+          install_command << "}"
+          install_command << render_command
+          install_command.join("\n\n")
```

Comment:
We would like to keep different parts of the script in different files @schisamo so that in the future we can add methods that can render only the part of the script. I think we can do a little bit of a better job here organizing things but I am against conversion to templates since we would like to keep things separate.